### ISSUE TITLE
useLayoutEffect warning in nextjs

### ISSUE THREAD

**Issue by SunnyAureliusRichard** [rpearce/react-expanding-textarea#81]
Hi,

Is it possible to have a setting that switches between useEffect and useLayoutEffect? I can see it was introduced in #77 .

This is the warning I'm getting:

    Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded into the server renderer's output format. This will lead to a mismatch between the initial, non-hydrated UI and the intended UI. To avoid this, useLayoutEffect should only be used in components that render exclusively on the client. See https://reactjs.org/link/uselayouteffect-ssr for common fixes.
        at ExpandingTextarea (web\node_modules\react-expanding-textarea\dist\cjs\index.js:54:27)
        at withForwardedRef(ExpandingTextarea)
        at Textarea (webpack-internal:///./src/components/atoms/inputs/Textarea.tsx:15:11)

Thanks!

**Comment by rpearce** [comment]
That's a bummer.

I'll see about adding this as a patch release when I get some time. It might be tomorrow or a few days from now.

```js
const useSSRLayoutEffect = typeof window === 'undefined'
  ? () => {}
  : useLayoutEffect
```

**Comment by rpearce** [comment]
This fix should be available in `v2.3.5`. I've also added you to the all contributors in the README for your bug reporting! Thank you.

<img width="364" alt="imagen" src="https://user-images.githubusercontent.com/592876/156494401-25a17bd0-2f05-4f1d-8284-f112915edb29.png">
